rfcs/0031-bazel.md

```diff
@@ -0,0 +1,358 @@
+  * develop `rules_ocaml`
+    * Option 1: Integrate Bazel and Dune
+      * Analyze `rules_foreign_cc` to see how it handles integration of
+    configure and make.
+        * Prototype a few Minimal Working Examples of using Bazel together with Dune.
+        * Implement `rules_ocaml` using this integrational strategy
+    * Option 2: Bazel-only ruleset
+      * Analyze `rules_go` (Bazel rules for Golang) as a possible model for `rules_ocaml`
+      * Implement `rules_ocaml` using this integrational strategy
+  * Use the new `rules_ocaml` to implement bazel support for Coda codebase:
+    * External libs
+    * Internal OCaml code
+    * C/C++ interop code
+* Golang
+  * Coda uses the Golang implementation of libp2p. Google maintains an
+    official set of Bazel rules for Go ([rules_go](https://github.com/bazelbuild/rules_go), so this should be
+    straightforward.
+
+**STATUS**
+
+* All of the external C/C++ libs have been bazelized, except one,
+  Postgresql, which actually builds correctly, but then fails in the
+  Bazel code.  I haven't figured out a fix yet; the workaround is to
+  just follow current practice and install Postgresql separately.
+* I've got some cross-compile code from a previous project that
+  supports Mac->Linux (using a crosstools-NG toolchain) and
+  Mac->Android (using the Android NDK toolchain).  Bazel has changed
+  its toolchain model, so these must be converted. I've done that with
+  the crosstools-NG toolchain and managed a minimal build of some of
+  the libraries with it.  It will require a little more work and
+  testing, but basically we're in good shape as far as C/C++ cross
+  compilation is concerned.
+* Haven't really looked into Bazel-Ocaml yet.
+
+### Effort Estimates and Timeline
+
+Effort = estimated remaining work effort in hours.
+
+C/C++
+
+| | Platform | Task | Effort | Due | Notes |
+| -- | ----------- | ----------- | ------ | ---------- | ---------------|
+| 1  | all | C/C++ external libs | 1 | Fri 5/15 | See note 1 |
+| 2 | all | libsnark | 3 | Fri 5/15 |
+| 3 | Mac to Linux x86 | cross-platform build | 4 | Fri 5/15 | Code adapted from previous project needs polish and testing |
+| 4 | Mac to Linux arm | cross-platform build | 2 | W 5/20 | Once an x86 toolchain is working adding support for other target archs should be easy |
+| 5 | Mac to Android | cross-platform build | 8 | W 5/20 |Reusing and adapting code from previous project |
+| 6 | linux | configure dev. env | 2 | Fri 5/22 |assuming a docker container can be used as a linux dev environment on Mac |
+| 7 | linux | C/C++ min. bld | 2 | Fri 5/22 |the Mac stuff should work with few changes |
+| 8 | windows | configure dev. env | 4 | TBD | setup VM image, install windows, install tools, etc. |
+| 9 | windows | C/C++ min. bld | 8 | TBD |paths, compiler flags, etc. will take some work, depending on toolchain |
+
+Notes:
+
+1. External libs:
+  * Assumption: what works on the mac will require little or no effort to work on Linux and Windows
```

Comment:
true for Linux, it will not work easily on Windows -- it's not worth looking into that for now.